packages/yourloops/components/switch-role/index.tsx

```diff
@@ -29,50 +29,57 @@
 import React from "react";
 import { useTranslation } from "react-i18next";
 import bows from "bows";
+import _ from "lodash";
 
+import { Profile } from "../../models/shoreline";
+import { HcpProfession } from "../../models/hcp-profession";
+import { SwitchRoleDialogsProps, SwitchRoleToHcpSteps } from "./models";
 import metrics from "../../lib/metrics";
 import { useAuth } from "../../lib/auth";
 import { useAlert } from "../utils/snackbar";
-import { SwitchRoleDialogsProps, SwitchRoleToHcpSteps } from "./models";
 import SwitchRoleConsequencesDialog from "./consequences-dialog";
 import SwitchRoleConsentDialog from "./consent-dialog";
+import SwitchRoleProfessionDialog from "./profession-dialog";
 
 const log = bows("SwitchRoleDialogs");
 
 function SwitchRoleDialogs(props: SwitchRoleDialogsProps): JSX.Element {
   const { t } = useTranslation("yourloops");
-  const { switchRoleToHCP, user } = useAuth();
+  const { switchRoleToHCP, updateProfile, user } = useAuth();
   const alert = useAlert();
   const [switchRoleStep, setSwitchRoleStep] = React.useState<SwitchRoleToHcpSteps>(SwitchRoleToHcpSteps.none);
+  const [feedbackConsent, setFeedbackConsent] = React.useState<boolean>(false);
 
   if (user === null) {
     throw new Error("User must be looged-in");
   }
 
   const role = user.role;
-  const handleSwitchRoleToConditions = (accept: boolean): void => {
-    const nextStep = accept ? SwitchRoleToHcpSteps.consent : SwitchRoleToHcpSteps.none;
-    setSwitchRoleStep(nextStep);
-    if (!accept) {
-      props.onCancel();
-    }
+  const handleSwitchRoleToConditions = (): void => {
+    setSwitchRoleStep(SwitchRoleToHcpSteps.consent);
   };
-  const handleSwitchRoleToUpdate = (accept: boolean, feedbackConsent: boolean): void => {
-    const nextStep = accept ? SwitchRoleToHcpSteps.update : SwitchRoleToHcpSteps.none;
-    setSwitchRoleStep(nextStep);
 
-    if (accept) {
-      switchRoleToHCP(feedbackConsent)
-        .then(() => {
-          metrics.send("switch_account", "accept_terms");
-        })
-        .catch((reason: unknown) => {
-          alert.error(t("modal-switch-hcp-failure"));
-          log.error("switchRoleToHCP", reason);
-        });
-    } else {
-      props.onCancel();
-    }
+  const getUpdatedProfile = (hcpProfession: HcpProfession): Profile => {
+    const updatedProfile = _.cloneDeep(user.profile ?? {}) as Profile;
+    updatedProfile.hcpProfession = hcpProfession;
+    return updatedProfile;
+  };
+
+  const handleSwitchRoleToUpdate = async (hcpProfession: HcpProfession): Promise<void> => {
+    await updateProfile(getUpdatedProfile(hcpProfession), false);
+    switchRoleToHCP(feedbackConsent)
```

Comment:
since it's already an async function, why not use a await ?

Reply:
And better, you should not update the profile with the profession first.
Since switchRoleToHCP() already update the profile, to update the accept terms date.
So you can simply change this function to:
```
const switchRoleToHCP = async (feedbackConsent: boolean, profession: string): Promise<void>
```
and pass directly the profession.

Reply:
suggestion:
```typescript
    try {
      await switchRoleToHCP(feedbackConsent, hcpProfession);
      metrics.send("switch_account", "accept_terms");
    } catch {
      alert.error(t("modal-switch-hcp-failure"));
      log.error("switchRoleToHCP", reason);
    }
```